simplecnf.go

```diff
@@ -0,0 +1,160 @@
+package go_sat_solver
+
+import "math"
+
+type orClause []VariableAssignment
+
+type simpleSatCnf struct {
+	clauses []orClause
+}
+
+func (f simpleSatCnf) IsSatisfied() bool {
+	return len(f.clauses) == 0
+}
+
+func (f simpleSatCnf) IsContradiction() bool {
+	for _, c := range f.clauses {
+		if len(c) == 0 {
+			return true
+		}
+	}
+	return false
+}
+
+type variableAssignment map[Variable]bool
+
+// putAll adds all the assignments to the map
+func (t variableAssignment) putAll(toAdd map[Variable]bool) {
+	m := map[Variable]bool(t)
+	for k, v := range toAdd {
+		m[k] = v
+	}
```

Comment:
shouldn't you return m? it seems like this is doing work and then leaving it there to be garbage collected or whatever.